ddl/ddl_api.go

```diff
@@ -2875,9 +2893,23 @@ func (d *ddl) TruncateTable(ctx sessionctx.Context, ti ast.Ident) error {
 		BinlogInfo: &model.HistoryInfo{},
 		Args:       []interface{}{newTableID},
 	}
+	if ok, _ := ctx.CheckTableLocked(tb.Meta().ID); ok {
+		// AddTableLock here to avoid this ddl job was execute successful but the session was been kill before return.
+		// The session will release its all table locks, if don't add table lock of the new table id here,
```

Comment:
`will release all table locks it holds, if we don't add the new locking table id here, the session may forget to release the new locked table id when this ddl job was executed successfully but the session was killed before return.`